.github/workflows/ci.yml

```diff
@@ -434,126 +335,26 @@ jobs:
         run: echo "run=true" >> $GITHUB_OUTPUT
         shell: bash
 
-      - uses: ./.github/actions/setup-rust
-        if: steps.rust-changes.outputs.run == 'true'
-        with:
-          version: ${{ env.rust-version }}
-          cache-key: integration
-          os: ${{ matrix.os }}
-        timeout-minutes: 10
-
-      # Pyo3 need additional configuration on MacOS to be manually build
-      # see https://pyo3.rs/v0.17.3/building_and_distribution.html#macos
-      - name: (🍎 macOS) Configure cargo for PyO3
-        if: >-
-          steps.rust-changes.outputs.run == 'true'
-          && startsWith(matrix.os, 'macos-')
-        shell: bash
-        run: |
-          set -eux
-          set -o pipefail
-          mkdir -p .cargo
-          cat << EOF | tee .cargo/config.toml
-          [target.x86_64-apple-darwin]
-          rustflags = [
-            "-C", "link-arg=-undefined",
-            "-C", "link-arg=dynamic_lookup",
-          ]
-          EOF
-
-      - name: Build rust test codebase
+      - uses: actions-rust-lang/setup-rust-toolchain@685abf830623a8456904376fce2de24302d98a7f # pin v1.3.4
         if: steps.rust-changes.outputs.run == 'true'
-        shell: bash
-        run: cargo test --workspace --profile ${{ env.CARGO_PROFILE }} ${{ env.cargo-exclude-unused-crates }} --no-run
```

Comment:
No way to keep it ?

That would be nice to not compile everything during the test

Reply:
nop, this is unneeded complexity:
- more code in the config file
- risk of having it wrong and doing needless compilation
- time of test vs time of compilation is already provided in the output of `cargo test`

Reply:
:shrug: